demos/python_demos/multi_camera_multi_person_tracking/multi_camera_multi_person_tracking.py

```diff
@@ -0,0 +1,146 @@
+        self.frames_queue = queue.Queue()
+        self.capture = capture
+        self.max_queue_length = max_queue_length
+
+    def __call__(self):
+        while self.process:
+            if self.frames_queue.qsize() > self.max_queue_length:
+                time.sleep(0.1)
+            has_frames, frames = self.capture.get_frames()
+            if not has_frames and self.frames_queue.empty():
+                self.process = False
+                break
+            if has_frames:
+                self.frames_queue.put(frames)
+
+
+def run(params, capture, detector, reid):
+    win_name = 'Multi camera tracking'
+    config = {}
+    if len(params.config):
+        config = read_py_config(params.config)
+
+    tracker = MultiCameraTracker(capture.get_num_sources(), reid, **config)
+
+    thread_body = frames_thread_body(capture, max_queue_length=len(capture.captures) * 2)
+    frames_thread = Thread(target=thread_body)
+    frames_thread.start()
+
+    if len(params.output_video):
+        video_output_size = (1920 // capture.get_num_sources(), 1080)
+        fourcc = cv.VideoWriter_fourcc(*'XVID')
+        output_video = cv.VideoWriter(params.output_video,
+                                      fourcc, 24.0,
+                                      video_output_size)
+    else:
+        output_video = None
+
+    while cv.waitKey(1) != 27 and thread_body.process:
+        start = time.time()
+        try:
+            frames = thread_body.frames_queue.get_nowait()
+        except queue.Empty:
+            frames = None
+
+        if frames is None:
+            continue
+
+        all_detections = detector.get_detections(frames)
+        all_masks = [[] for _ in range(len(all_detections))]
+        for i, detections in enumerate(all_detections):
+            all_detections[i] = [det[0] for det in detections]
+            all_masks[i] = [det[2] for det in detections if len(det) == 3]
+
+        tracker.process(frames, all_detections, all_masks)
+        tracked_objects = tracker.get_tracked_objects()
+
+        fps = round(1 / (time.time() - start), 1)
+        vis = visualize_multicam_detections(frames, tracked_objects, fps)
+        cv.imshow(win_name, vis)
+        if output_video:
+            output_video.write(cv.resize(vis, video_output_size))
+
+    thread_body.process = False
+    frames_thread.join()
+
+    if len(params.history_file):
+        history = tracker.get_all_tracks_history()
+        with open(params.history_file, 'w') as outfile:
+            json.dump(history, outfile)
+
+
+def main():
+    """Prepares data for the person recognition demo"""
+    parser = argparse.ArgumentParser(description='Multi camera multi person \
+                                                  tracking live demo script')
+    parser.add_argument('--videos', type=str, nargs='+', help='Input videos')
+    parser.add_argument('--cam_ids', type=int, nargs='+', default=[-1],
+                        help='Indexes of input cameras')
```

Comment:
I think `--videos` and `--cam_ids` can be merged into a single key `-i`. The only difference between them is that for camera some properties are set. You probably can differentiate what the input was by checking if setting or getting some `VideoCapture` property failed.

Reply:
Fixed